src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/SaveTempDataPropertyFilterBase.cs

```diff
@@ -98,10 +98,18 @@ private static void ValidateProperty(PropertyHelper propertyHelper)
 
             var propertyType = Nullable.GetUnderlyingType(property.PropertyType) ?? property.PropertyType;
 
-            if (!(propertyType.GetTypeInfo().IsPrimitive || propertyType == typeof(string)))
+            if (!TempDataSerializer.CanSerializeType(propertyType, out var actualType, out var errorMessage))
```

Comment:
Under what conditions will this return false with a null error message? Should we just solve that?

Reply:
Good point. I fixed this one now.